### PR TITLE
isPartiallyOnScreen method

### PR DESCRIPTION
**Pull request by keyurpatel**
Close https://github.com/kangax/fabric.js/issues/4823,

Issue:  to know if the object is partially off screen, completely off screen.

Solution: create a simple helper methods which return boolean for object partially off screen `isPartiallyOnScreen`, completely offscreen `isOffScreen`.

### CODE REVIEW

File: src/mixins/object_geometry.mixin.js [hunk 1]
```diff
@@ -187,6 +187,35 @@
       return false;
     },
 
+    /**
+     * Checks if object is partially contained within the canvas with current viewportTransform
+     * the check is done stopping at first point that appears off screen
+     * @param {Boolean} [calculate] use coordinates of current position instead of .oCoords
```
Comment (asturur) [inline src/mixins/object_geometry.mixin.js]: use coordinates of current position, calculated just in time, instead of precalculated .aCoords

Comment (asturur) [reply src/mixins/object_geometry.mixin.js]: apply to the other function too.

File: test/unit/object_geometry.js [hunk 2]
```diff
@@ -802,4 +802,38 @@
...
+    assert.equal(cObj.isPartiallyOnScreen(cObj.aCoords), true, 'object is partially on screen because it include the canvas');
+    cObj.left = 100;
+    cObj.top = 100;
+    cObj.setCoords();
+    assert.equal(cObj.isPartiallyOnScreen(cObj.aCoords), false, 'object is not partially on screen because it include the canvas');
+    cObj.left = -1000;
+    cObj.top = -1000;
+    cObj.setCoords();
+    assert.equal(cObj.isPartiallyOnScreen(cObj.aCoords), false, 'object is not partially on screen because it is outside the canvas');
+  });
```
Comment (asturur) [inline test/unit/object_geometry.js]: we have more test to cover and edge cases i believe. Intersect can return true for coincidents segments too.
